### ISSUE TITLE
Persistent data not restored.

### ISSUE THREAD

**Issue by coconutlulz** [mher/flower#476]
Running flower (ae19ec5f) with patch from https://github.com/mher/flower/issues/429 and a few extra debug log statements.

`--persistent` flag is used. Celery version v3.1.18.

If I create a couple of tasks, they run as expected.

After I send a `SIGINT`:

`[D 150923 14:43:09 events:96] Saving state to 'flower'...`
`[D 150923 14:43:09 events:97] <State: events=54 tasks=4>`

The DB file 'flower' clearly contains the correct data. When I start flower again:

`[D 150923 14:47:35 events:76] Loading state from 'flower'...`
`[D 150923 14:47:35 events:80] <State: events=0 tasks=0>`

If I run Python and load the file with shelve:

`>>> f['events']`
`<State: events=0 tasks=0>`

So, something isn't working correctly when shelve reads the file.

**Comment by nickabbey** [comment]
I have the same behavior, though I haven't yet attempted to load the file with shelve

**Comment by nickabbey** [comment]
just verified same behavior in shell:
In [2]: shelve.open('/.run/flower_db')
Out[2]: {'events': <State: events=0 tasks=0>}

**Comment by nickabbey** [comment]
Looks like this may be a lapse in expectation, or communication/documentation of behavior? 
I see that the "main" page with the counts for the individual workers is reset to all zeroes every time, as is the graph. As I understand it, the graph is not supposed to persist. However, I would have expected the counts on the front page to persist. 
That said, however, I see that if I go to the tasks page, the history exists there as expected. 
Perhaps the documentation should be clarified to set expectations correctly?
IE) Specifically call out in the documentation that only the /tasks/ page lists persistent results while the counts on the front page and the graphs are not persisted by design.

I'll note that while I took numerous steps to address this perceived issue, including updating from 0.8.3 to master, adding 
`stopsignal=TERM`
to my supervisord config
and specifying a path for the file with 
`db=/path/to/file`
none were actually necessary to get persistence working. 

Once I realized that I was looking for the wrong thing, I reverted my config back to it's original state (when the issue was reported here), uninstalled flower and all dependencies, then reinstalled them with 
`pip install flower`

The tasks page continues to show all persistent task results as expected.
